ReactiveCocoa/Swift/SignalProducer.swift

```diff
@@ -475,10 +475,10 @@ public func concat<T, E>(producer: SignalProducer<SignalProducer<T, E>, E>) -> S
 				serialDisposableCompositeHandle.remove()
 				state.modify { (var state) in
 					state.latestSignalCompleted = true
-					let outerSignalIsComplete = completeIfAllowed(state)
-					if !outerSignalIsComplete {
-						nextSignalProducer = state.queuedSignalProducers[0]
-						state.queuedSignalProducers.removeAtIndex(0)
+					if state.queuedSignalProducers.isEmpty {
+						completeIfAllowed(state)
```

Comment:
I know this wasn't from your changes, but can you refactor this operator so that all invocations of `completeIfAllowed` occur _outside_ of the `modify` block? Otherwise, we'll be holding the spinlock for way longer than intended, and could even deadlock.